packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -58,41 +63,77 @@ class ${class_name}:
         Args:
             discovery_client: An optional discovery client.
 
+            pin_map_client: An optional pin map client.
+
             grpc_channel: An optional gRPC channel targeting a measurement service.
 
             grpc_channel_pool: An optional gRPC channel pool.
         """
-        self._initialization_lock = threading.Lock()
+        self._initialization_lock = threading.RLock()
```

Comment:
why are we using Reentrant lock here?

Reply:
To create other clients, we use separate locks, which can lead to deadlocks when calling methods from here like `_get_discovery_client`. Using RLock in this context allows the same thread to release a previously acquired lock if necessary.